clippy_lints/src/operators/float_cmp.rs

```diff
@@ -105,6 +113,78 @@ fn is_allowed(val: &Constant<'_>) -> bool {
     }
 }
 
+// This is a best effort guess and may have false positives and negatives.
+fn is_pure_expr<'tcx>(cx: &LateContext<'tcx>, e: &'tcx Expr<'_>) -> bool {
```

Comment:
What are you trying to check here exactly? What is a "pure expression"?

Reply:
No side effects.

Reply:
Then please add a comment explaining that on the function.

Reply:
Functional purity isn't a weird concept. It's also mentioned casually in comments a few times in other places.